aspnetcore/security/authentication/identity-configuration.md

```diff
@@ -2,22 +2,86 @@
+
+`IdentityOptions.Lockout` has the following properties:
+* `DefaultLockoutTimeSpan`: The amount of time a user is locked out when a lockout occurs. Defaults to 5 minutes.
+* `MaxFailedAccessAttempts`: The number of failed access attempts until a user is locked out, if lockout is enabled. Defaults to 5.
+* `AllowedForNewUsers`: Determines if a new user can be locked out. Defaults to true.
 
-[!code-csharp[Main](identity/sample/src/ASPET-IdentityDemo-PrimaryKeysConfig/Startup.cs?highlight=2&range=60-65)]
+
+## Sign in settings
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,44-46,50-52)]
+
+`IdentityOptions.SignIn` has the following properties:
+* `RequireConfirmedEmail`: Requires a confirmed email to sign in. Defaults to false.
+* `RequireConfirmedPhoneNumber`: Requires a confirmed phone number to sign in. Defaults to false.
+
+
+## User validation settings
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,48-52)]
+
+`IdentityOptions.User` has the following properties:
+* `RequireUniqueEmail`: Requires each User to have a unique email. Defaults to false.
+* `AllowedUserNameCharacters`: Allowed characters in the username. Defaults to abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+.
 
 ## Application's cookie settings
 
-Like the passwords policy, all the settings of the application's cookie can be changed in the startup class.
+Like the passwords policy, all the settings of the application's cookie can be changed in the `Startup` class.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+Under `ConfigureServices` in the `Startup` class, you can configure the application's cookie.
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?name=snippet_configurecookie)]
+
+`CookieAuthenticationOptions` has the following properties:
+* `Cookie.Name`: The name of the cookie.  Defaults to .AspNetCore.Cookies.
+* `Cookie.HttpOnly`: When true, the cookie is not accessible from client-side scripts. Defaults to true.
+* `ExpireTimeSpan`: Controls how much time the authentication ticket stored in the cookie will remain valid from the point it is created. Defaults to 14 days.
+* `LoginPath`: When a user is unauthorized, they will be redirected to this path to login. Defaults to /Account/Login.
+* `LogoutPath`: When a user is logged out, they will be redirected to this path. Defaults to /Account/Logout.
+* `AccessDeniedPath`: When a user fails an authorization check, they will be redirected to this path. Defaults to /Account/AccessDenied.
+* `SlidingExpiration`: When true, a new cookie will be issued with a new expiration time when the current cookie is more than halfway through the expiration window. Defaults to true.
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 [!code-csharp[Main](identity/sample/src/ASPET-IdentityDemo-PrimaryKeysConfig/Startup.cs?highlight=2&range=72-80)]
```

Comment:
- Change this folder name to "ASPNET-IdentityDemo-PrimaryKeysConfig" and update the code snippet import paths accordingly.
- This 1.x code snippet should display the `services.Configure` portion of the code as well, to provide a bit more context to the reader. I'd expect the snippet to display like this:
```csharp
services.Configure<IdentityOptions>(options =>
{
    // Cookie settings
    options.Cookies.ApplicationCookie.CookieName = "YourAppCookieName";
    options.Cookies.ApplicationCookie.ExpireTimeSpan = TimeSpan.FromDays(150);
    options.Cookies.ApplicationCookie.LoginPath = "/Account/LogIn";
    options.Cookies.ApplicationCookie.LogoutPath = "/Account/LogOff";
    options.Cookies.ApplicationCookie.AccessDeniedPath = "/Account/AccessDenied";
    options.Cookies.ApplicationCookie.AutomaticAuthenticate = true;
    options.Cookies.ApplicationCookie.AuthenticationScheme = Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationDefaults.AuthenticationScheme;
    options.Cookies.ApplicationCookie.ReturnUrlParameter = Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationDefaults.ReturnUrlParameter;
});
```